### DISCUSSION TITLE
Dynamic Filter on Data Provider

### DISCUSSION THREAD

**Discussion by RedBlob** [Budibase/budibase#10727] [Help] [answered]
Hello, 
I am using Postgres database as a datasource and I nave added a dynamic filters to a table block.
The table block is directly connected with a table on PostgreSQL.
The data on the table block is managed correctly and the sort order is OK.
The table block shown only 4 columns of 10 present in the PostgreSQL table and the scroll limit is limited to 50 row.
All is working OK.
The problem is the Dynamic Filter ...... Whatever setting is activated in the filter ( less than , equal to , greater than ) the data in the table are not changed.
It is as if the filter has no influence on the data represented in the table,

Is it an issue or I am doing something wrong?

Thank You
Best Regards

**Comment by ConorWebb96** [comment]
Hey @RedBlob,

What is the value of activated? Is this a Boolean field? If so I'm unsure why you would ever use less than or greater to. Wouldn't this specifically be equal to true/false?

I've created and example of this working, below are various screenshots using the table component with the dynamic filter, with the end result showing it filtering on the boolean value.

**Data used/columns in the table**
<img width="789" alt="Screenshot 2023-05-30 at 08 07 24" src="https://github.com/Budibase/budibase/assets/126772285/0e3e923f-15c7-4fc9-b3c7-1bca78d54e22">

**Component tree + output of the table**
<img width="1468" alt="Screenshot 2023-05-30 at 08 09 23" src="https://github.com/Budibase/budibase/assets/126772285/d9e5d3aa-63d0-46f8-b172-5d79731e92a8">

**Working demo showing filtering in action.**
![dynamic filter on boolean](https://github.com/Budibase/budibase/assets/126772285/91e75dda-9884-4fd6-b9c9-cd1a96a5b3a9)

I'm currently running the above on Budibase: 2.6.20.

Hopefully this helps!

**Reply by RedBlob** [reply]
Hello @ConorWebb96 ,

Thank you for Help but it seems your example work OK only if the database is local.

If You use this configuration on a PostgreSQL table it don't work.

The column I use are all integer and I use the filter "less than" or "less than" applied to numbers ( integer from 1 to 1000 ).

It seems the filter don' work on external table ( the data provider is an external table ).

Can you check your example with an external PostgreSQL table like this 

CREATE TABLE IF NOT EXISTS public.mytable
(
    id_fex bigint NOT NULL DEFAULT nextval('id_fex_seq'::regclass),  
    largh integer NOT NULL DEFAULT 0,
    alte integer NOT NULL DEFAULT 0,
    lungh integer NOT NULL DEFAULT 0,
    CONSTRAINT local_flux_pkey PRIMARY KEY (id_fex)
)

Thank You

**Reply by melohagan** [reply]
Hey @RedBlob Could you provide some screenshots of the filters you are setting in your Dynamic Filter please, as well as the result in the table.

**Reply by ConorWebb96** [reply]
Hey @RedBlob,

Apologies, for the above example I should have clarified, I was using the Postgres data source. Below is the sql used to create the table.

```
CREATE TABLE details (
  id SERIAL PRIMARY KEY,
  title VARCHAR(255),
  description TEXT,
  activated BOOLEAN
);
```

<img width="1047" alt="Screenshot 2023-06-01 at 07 44 10" src="https://github.com/Budibase/budibase/assets/126772285/f4df30aa-fc2a-4b06-a594-2aafef323bef">


Hopefully this helps.

**Comment by RedBlob** [comment]
Hi @ConorWebb96 and @melohagan ,

It is probably something simple that we can't set ( very likely ) or there is a bug.

I created a table with this SQL 

CREATE TABLE IF NOT EXISTS public.elenco_destinatari
(
    id_dest bigint NOT NULL DEFAULT nextval('id_dest_seq'::regclass),
    nome_societa text COLLATE pg_catalog."default" NOT NULL DEFAULT 'x'::text,
    indirizzo_impianto text COLLATE pg_catalog."default" NOT NULL DEFAULT 'x'::text,
    citta_impianto text COLLATE pg_catalog."default" NOT NULL DEFAULT 'x'::text,
    provincia_impianto text COLLATE pg_catalog."default" NOT NULL DEFAULT 'x'::text,
    cap_impianto text COLLATE pg_catalog."default" NOT NULL DEFAULT 'x'::text,
    data_evento_update timestamp without time zone NOT NULL DEFAULT '2000-01-01 00:00:00'::timestamp without time zone,
    CONSTRAINT local_dest_pkey PRIMARY KEY (id_dest)
)

TABLESPACE pg_default;

ALTER TABLE IF EXISTS public.elenco_destinatari
    OWNER to postgres;

GRANT SELECT, UPDATE, INSERT, DELETE ON TABLE public.elenco_destinatari TO redblob;

All is OK 

Now I have setup a Table block to show the data in the table :

![Immagine 2023-06-01 152544](https://github.com/Budibase/budibase/assets/70526244/d92180ea-f363-499e-8178-9a1150bfa3e4)

All is OK when I try to show the data.

Now I ask to setup a filter to select only one condition 

![Immagine 2023-06-01 152809](https://github.com/Budibase/budibase/assets/70526244/e207543d-abf7-4396-9b5c-401ab84959fa)

When I confirm I should have only one record as result ....... but the result is this 

![Immagine 2023-06-01 152947](https://github.com/Budibase/budibase/assets/70526244/4de18150-407d-40a4-80ff-0a3d81eef752)

Exactly like before.

I have tested also the condition  "START WITH " ----> AZIENDA
I have tested also the condition  "EQUAL" ---- "AZIENDA01 SRL"
I have tested also the condition  "IS EMPTY" 

The result is always the same .......
This is very strange ..... what I have wrong ?

Thank You

**Reply by melohagan** [reply, marked as answer]
Hey @RedBlob 

I think I understand what is wrong now.

The **Dynamic filter** is designed to be used with a **Data provider**. Therefore you cannot use it with a **Table block** as you can see in this screenshot:

![Screenshot 2023-06-01 at 16 38 21](https://github.com/Budibase/budibase/assets/101575380/c6058de9-d318-40a1-8e3e-2b454c4ac340)

Instead you will need to **Eject** the table block, and then you can use the dynamic filter:
![Screenshot 2023-06-01 at 16 39 46](https://github.com/Budibase/budibase/assets/101575380/450875ff-9d98-4624-a8b4-9f1c16f9f752)

With that said, if you are using a Table block, then you can add *Search fields* within the Settings Panel:

![Screenshot 2023-06-01 at 16 42 04](https://github.com/Budibase/budibase/assets/101575380/39589f07-9e2d-46c8-83ef-9d5795320f2d)

**Comment by RedBlob** [comment]
Hi @melohagan ,

Bingo !

You are right !

If I use the **Table**  instead of **Table Block** the Dynamic filter work correctly.

Thank You for help ....... and for your patience !

Best Regards

**Reply by melohagan** [reply]
No problem, glad you got it working!

**Comment by ChingyamYeung7** [comment]
Now it is june 28th 2025, there is no ejection for the table, and I used all the steps above but I could not search the field for table. there is no such option. When I use filter it doesn't work. what should I do? 
![image](https://github.com/Budibase/budibase/assets/83433070/f4f63198-08cf-4cd7-851c-79d9adaf1fa9)

**Reply by LucianHorrobin** [reply]
Also can't seem to get search as demonstrated in the docs or dynamic filter to work.
